conans/client/settings_preprocessor.py

```diff
@@ -0,0 +1,18 @@
+from conans.errors import InvalidConanSettingField
+
+
+def preprocess(settings, output):
+    fill_runtime(settings, output)
+
+
+def fill_runtime(settings, output):
+    try:
+        if settings.compiler == "Visual Studio":
+            if settings.get_safe("compiler.runtime") is None:
+                settings.compiler.runtime = "MDd" if settings.get_safe("build_type") == "Debug" \
+                                                  else "MD"
+                output.info("Setting 'compiler.runtime' not declared, automatically "
+                            "adjusted to '%s'" % settings.compiler.runtime)
+    except InvalidConanSettingField:  # If the settings structure doesn't match these general
```

Comment:
The usage of this Exception type seems a bit weak to me. I would have used just safe checks of settings, no Exception capture: Just an opinion, not really sure about best approach.